webapp/src/components/kanban/kanban.tsx

```diff
@@ -55,7 +56,17 @@ const vStrength = createVerticalStrength(Utils.isMobile() ? 60 : 250)
 
 const Kanban = (props: Props) => {
     const currentView = useAppSelector(getCurrentView)
+    const cardTemplates: Card[] = useAppSelector(getCurrentBoardTemplates)
     const {board, activeView, cards, groupByProperty, visibleGroups, hiddenGroups, hiddenCardsCount} = props
+    const [defaultTemplateID, setDefaultTemplateID] = useState<string>()
+
+    useEffect(() => {
+        if(currentView && currentView.fields && currentView.fields.defaultTemplateId) {
+            if(cardTemplates.find(ct => ct.id === currentView.fields.defaultTemplateId)) {
+                setDefaultTemplateID(currentView.fields.defaultTemplateId)
+            }
+        }
+    }, [currentView])
```

Comment:
Since you are only using `currentView.fields.defaultTemplateId` within the use effect. That should be your dependency. If other fields within `currentView` or `currentView.fields` change, we don't need to rerun this effect, only if `currentView.fields.defaultTemplateId` changes. Sorry missed that earlier. 

```suggestion
    useEffect(() => {
        if(currentView.fields.defaultTemplateId) {
            if(cardTemplates.find(ct => ct.id === currentView.fields.defaultTemplateId)) {
                setDefaultTemplateID(currentView.fields.defaultTemplateId)
            }
        }
    }, [currentView.fields.defaultTemplateId])
```

Reply:
Yes, the dependency must be changed, but the condition in the if statement should remain the same because the component `kanban` get mounted on the application when it runs for the first without any board, that time the value of `currentView` will be `undefined` which will result into the error `Cannot access the property fields of undefined`, that the reason why we need to check all the 3 values.

Reply:
I have made some more changes because it was failing the git CI. Please have a look, thanks.

Reply:
I guess we need to keep it the same way because whenever the application is loaded for the first time, the component `kanban` will be mounted and the value of the `currentView` will be `undefined` and it will throw an error `cannot find the property fields of undefined.`. And I guess the dependency will not have an impact because the `defaultTemplateView` will only change if the `currentView` changes.  I have tried your suggested changes but it seems that some test case of cypress was failing due to `cannot read the property fields of undefined`. Thoughts.